scripts/gitlab/hatchet-analysis.py

```diff
@@ -22,32 +23,100 @@
 
 import hatchet as ht
 
-f1 = options.report
-f2 = options.baseline
-
-gf1 = ht.GraphFrame.from_caliperreader(f1)
-gf2 = ht.GraphFrame.from_caliperreader(f2)
-
-gf3 = gf2 - gf1
-
-# Display dataframe columns
-print(gf3.dataframe.columns)
+class GenericFrame(ht.GraphFrame):
+   def __init__(self, gf):
+      generic_dataframe = gf.dataframe.copy()
+      generic_graph = gf.graph.copy()
+      generic_exc_metrics = gf.exc_metrics
+      generic_inc_metrics = gf.inc_metrics
+      generic_default_metric = gf.default_metric  # in newer Hatchet
```

Comment:
No they are used in the super call
super().__init__(generic_graph, generic_dataframe, generic_exc_metrics, generic_inc_metrics)

Are you talking about the default metric?  We don't use that in ExtractCommonSubtree anymore since the metric can take one of two forms

#metric = "sum#inclusive#sum#time.duration"
metric = "Min time/rank"

The first is legacy metric which still  exists BTW within Caliper and is used in all of the timing calculations, while the second form is a Human readable alias but processed as first class citizen within Hatchet

But we keep the Default metric as a class variable to maintain self similar structure to the inherited Graphframe.